### PR TITLE
Improve bootloader version checks and add an option to toggle net-install-at-power-on

### PR DESCRIPTION
**Pull request by timg236**
* Allow the possibility of a downgrade e.g. switching from a pre-release back to mainline
* Prompt if the update will result in a downgrade
* Refactor the code to use a common bootloader_filename function
* Add a UI to toggle network-install-at-power-on so that this is available before the firmware release hits mainline.

### CODE REVIEW

**Comment by timg236** [comment]
@peterharperuk @XECDesign - In addition to the above this change supports removing all firmware releases older than the last automatic update from the rpi-eeprom APT package (moved to the old release dir in github)

File: raspi-config [hunk 1]
```diff
@@ -1878,6 +1878,76 @@ EOF
...
+      B2*)
+         # NET_INSTALL_AT_POWER_ON default value is 0
+         true
+         ;;
+      *)
+        whiptail --msgbox "Programmer error, unrecognised boot option" 20 60 2
+        return 1
+        ;;
+    esac
+    rpi-eeprom-config --apply $EECFG $FIL
```
Comment (lurch) [inline raspi-config]: Is the `$FIL` here supposed to be `$FILNAME` ?

Comment (timg236) [reply raspi-config]: Yes. Fixed.

**Comment by peterharperuk** [comment]
Looks ok

**Comment by timg236** [comment]
@spl237 - Is this ok to merge? The tweak to the version checks will make it easier to ditch old firmware releases from the rpi-eeprom packages.

**Comment by spl237** [comment]
As long as you're happy with it!

**Comment by timg236** [comment]
Thanks, works for me :)

**Comment by timg236** [comment]
Yes please

**Pull request merged**